packages/js/data/src/crud/types.ts

```diff
@@ -13,3 +22,79 @@ export type Item = {
 export type ItemQuery = BaseQueryParams & {
 	[ key: string ]: unknown;
 };
+
+export type CrudActions< ResourceName, ItemConfig, MutableProperties > =
+	MapActions<
+		{
+			create: ( query: ItemQuery ) => Item;
+			update: ( query: ItemQuery ) => Item;
+		},
+		ResourceName,
+		MutableProperties,
+		ItemConfig
+	> &
+		MapActions<
+			{
+				delete: ( id: IdType ) => Item;
+			},
+			ResourceName,
+			IdType,
+			ItemConfig
+		>;
+
+export type CrudSelectors<
+	ResourceName,
+	PluralResourceName,
+	ItemConfig,
+	ItemQueryConfig,
+	MutableProperties
+> = MapSelectors<
+	{
+		'': WPDataSelector< typeof getItem >;
+		Error: WPDataSelector< typeof getItemError >;
+		DeleteError: WPDataSelector< typeof getItemDeleteError >;
+		UpdateError: WPDataSelector< typeof getItemUpdateError >;
+	},
+	ResourceName,
+	IdType,
+	unknown
```

Comment:
Should this be `ItemConfig` or the `'': WPDataSelector< typeof getItem >;` be separated into it's own map, just like the `getItems` selector below?

As currently the `getProductAttribute( 2 )` returns a `unknown` type, but the `getProductAttributes({})` returns the correct `ProductAttribute[]` type. It be nice if that one can also return `ProductAttribute`.

Reply:
Great catch!  Updated.